samples/subsys/usb/dfu-next/README.rst

```diff
@@ -0,0 +1,137 @@
+.. zephyr:code-sample:: dfu-next
+   :name: USB DFU
+   :relevant-api: usbd_api usbd_dfu
+
+   Implement a basic USB DFU device
+
+Overview
+********
+
+This sample application demonstrates the USB DFU implementation using the
+new experimental USB device stack.
+
+Requirements
+************
+
+This project requires an experimental USB device driver (UDC API) and uses the
+:ref:`disk_access_api` and RAM-disk to download/upload the image.
+
+Building and Running
+********************
+
+This sample can be built for multiple boards, in this example we will build it
+for the reel board:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/usb/dfu-next
+   :board: reel_board
+   :goals: build flash
+   :compact:
+
+`dfu-util`_ tool can be used to download or upload the images. There are two
+modes of operation in the USB DFU, runtime and DFU. The example starts in
+runtime mode. To switch to DFU mode without uploading or downloading, the
+following command can be used:
+
+.. code-block:: console
+
+   dfu-util --detach
+
+Use the following command to upload the ``ramdisk0`` image to the host:
+
+.. code-block:: console
+
+   dfu-util --alt 0 --upload ramdisk0_backup.bin
+
+Use the following command to download the ``ramdisk0`` image to the device:
+
+.. code-block:: console
+
+   dfu-util --alt 0 --download ramdisk0_backup.bin
+
+Building with flash backend enabled
+***********************************
+
+The USB DFU device support has a built-in flash backend. This backend uses
+:ref:`flash_img_api` and :ref:`flash_map_api` to write or read flash image, the
+implementation is similar to the one we had in the previous USB DFU device
+example.
+
+To use flash backend set the :kconfig:option:`CONFIG_APP_USB_DFU_USE_FLASH_BACKEND`.
+An additional interface will be available in DFU mode to upload/download the
+SLOT-1 image.
+
+It is also possible to try the sample together with the MCUboot bootloader
+library. The following example shows how to build MCUboot and this sample with
+flash backend and MCUboot support enabled using the :ref:`sysbuild`:
+
+.. zephyr-app-commands::
+   :tool: west
+   :zephyr-app: samples/subsys/usb/dfu-next
+   :board: reel_board
+   :goals: build flash
+   :west-args: --sysbuild
+   :gen-args: -DSB_CONFIG_BOOTLOADER_MCUBOOT=y -DCONFIG_APP_USB_DFU_USE_FLASH_BACKEND=y
+
+Another application image is required to be used as a firmware update and
+downloaded to SLOT-1. Build and sign a second application image e.g.
+:zephyr:code-sample:`hello_world`, which will be used as an image for the
+update. Do not forget to enable the required :kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT`
+option (as described in :ref:`mcuboot`). For example:
+
+.. code-block:: console
+
+   west build -b reel_board zephyr/samples/hello_world -d build-hello_world -- \
+   -DCONFIG_BOOTLOADER_MCUBOOT=y \
+   '-DCONFIG_MCUBOOT_SIGNATURE_KEY_FILE="bootloader/mcuboot/root-rsa-2048.pem"'
```

Comment:
just noticed but this should use zephyr-app-commands (unless there is a reason you couldn't but doesn't look like it)
can be addressed as a follow-up

Reply:
I copied it from the old sample, I guess it was used because the gen-args line would be to long, but something like that seems to work.

```diff
-.. code-block:: console
-
-   west build -b reel_board zephyr/samples/hello_world -d build-hello_world -- \
-   -DCONFIG_BOOTLOADER_MCUBOOT=y \
-   '-DCONFIG_MCUBOOT_SIGNATURE_KEY_FILE="bootloader/mcuboot/root-rsa-2048.pem"'
+.. zephyr-app-commands::
+   :app: zephyr/samples/hello_world
+   :board: reel_board
+   :gen-args: -DCONFIG_MCUBOOT_SIGNATURE_KEY_FILE=\"bootloader/mcuboot/root-rsa-2048.pem\"
+              -DCONFIG_BOOTLOADER_MCUBOOT=y
+   :goals: flash
```
?

Reply:
Yeah, this should be the way to have directive options span on multiple lines but it looks like the \n between the two options is being kept in the output so there might be an issue in the zephyr-app-commands code. Like I said, for a follow-up anyway (although tbh I wouldn't worry about going over 100 characters line limit in the case of sphinx "code")

Reply:
I had to push anyway, so I changed it.